axelrod/tests/unit/test_tournament.py

```diff
@@ -394,26 +412,58 @@ def test_worker(self):
         queue_stop = done_queue.get()
         self.assertEqual(queue_stop, 'STOP')
 
-    def test_play_round_robin_mutable(self):
+    def test_play_matches(self):
         tournament = axelrod.Tournament(
             name=self.test_name,
             players=self.players,
             game=self.game,
             turns=200,
             repetitions=self.test_repetitions)
-        output = tournament._play_round_robin()
-        self.assertEqual(output['payoff'], self.expected_payoff)
-        self.assertTrue(
-            (axelrod.Cooperator, axelrod.Defector) in
-            tournament.deterministic_cache)
-
-    def test_play_round_robin_immutable(self):
+        matches = {
+            (0, 0): axelrod.Match((axelrod.Cooperator(), axelrod.Cooperator()), turns=200),
+            (0, 1): axelrod.Match((axelrod.Cooperator(), axelrod.TitForTat()), turns=200),
+            (0, 2): axelrod.Match((axelrod.Cooperator(), axelrod.Defector()), turns=200),
+            (0, 3): axelrod.Match((axelrod.Cooperator(), axelrod.Grudger()), turns=200),
+            (0, 4): axelrod.Match((axelrod.Cooperator(), axelrod.GoByMajority()), turns=200),
+            (1, 1): axelrod.Match((axelrod.TitForTat(), axelrod.TitForTat()), turns=200),
+            (1, 2): axelrod.Match((axelrod.TitForTat(), axelrod.Defector()), turns=200),
+            (1, 3): axelrod.Match((axelrod.TitForTat(), axelrod.Grudger()), turns=200),
+            (1, 4): axelrod.Match((axelrod.TitForTat(), axelrod.GoByMajority()), turns=200),
+            (2, 2): axelrod.Match((axelrod.Defector(), axelrod.Defector()), turns=200),
+            (2, 3): axelrod.Match((axelrod.Defector(), axelrod.Grudger()), turns=200),
+            (2, 4): axelrod.Match((axelrod.Defector(), axelrod.GoByMajority()), turns=200),
+            (3, 3): axelrod.Match((axelrod.Grudger(), axelrod.Grudger()), turns=200),
+            (3, 4): axelrod.Match((axelrod.Grudger(), axelrod.GoByMajority()), turns=200),
+            (4, 4): axelrod.Match((axelrod.GoByMajority(), axelrod.GoByMajority()), turns=200),
+        }
+        results = tournament._play_matches(matches)
+        self.assertNotIn('matches', results)
+        self.assertEqual(results['payoff'], self.expected_payoff)
+
+    def test_keep_matches(self):
         tournament = axelrod.Tournament(
             name=self.test_name,
             players=self.players,
             game=self.game,
-            turns=200,
-            repetitions=self.test_repetitions)
-        output = tournament._play_round_robin(cache_mutable=False)
-        self.assertEqual(output['payoff'], self.expected_payoff)
-        self.assertEqual(tournament.deterministic_cache, {})
+            turns=3,
+            repetitions=self.test_repetitions,
+            keep_matches=True)
+        matches = {
```

Comment:
Maybe parameterize `matches` with `turns=200` above and `turns=3` here in a function?